### ISSUE TITLE
Factory generation misnames output class

### ISSUE THREAD

**Issue by mattwellss** [bitExpert/phpstan-magento#263]
When working with a class that have "Factory" in its classname, such as `Company\Module\Model\FactoryThing`, the PHPStan-generated output class for its factory (`Company\Module\Model\FactoryThingFactory` is `Company\Module\Model\Thing`.

It's caused by a simple error in factory generation
https://github.com/bitExpert/phpstan-magento/blob/8e55592f511bce6eb8234dd38ecc451bb49bbe35/src/bitExpert/PHPStan/Magento/Autoload/FactoryAutoloader.php#L60

A more correct transformation:
```php
        $originalClassname = preg_replace('#Factory$#', '', $factoryClassname);
```

This way, only "Factory" at the end of the classname is removed.

**Comment by shochdoerfer** [comment]
Good catch, the other autoloaders might also be affected. Mind providing a PR to fix it?

**Comment by mattwellss** [comment]
@shochdoerfer It doesn't look to me like any other autoloaders are similarly affected. Regardless, I've got a fix ready for Factory autoloading, which resolves this issue.

**Comment by shochdoerfer** [comment]
Thanks! Will have a look later and merge. Very much appreciated.

**Comment by shochdoerfer** [comment]
Fixed with the latest 0.23.1 bugfix release. Thanks for helping out!
